code/game/gamemodes/game_mode.dm

```diff
@@ -236,9 +236,14 @@
 
 		for(var/mob/Player in GLOB.alive_mob_list)
 			if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player) && Player.client)
-				if(Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging!
-					living_antag_player = Player
-					return 0
+				if(Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging but is their antagonist datum important enough to skip mulligan?
+					var/prevent_roundtype_conversion = FALSE
+					for(var/datum/antagonist/antag_types in Player.mind.antag_datums)
+						if(antag_types.prevent_roundtype_conversion)
+							prevent_roundtype_conversion = TRUE

```

Comment:
Probably doesn't matter much, but couldn't you break here for sweet performance gains?

Reply:
It's also gross to call it antag_types when it's just one antag datum

Reply:
people can have multiple antag datums (brainwashed?), that's why the mind var is called antag_datums

Reply:
but yes the break is a great idea